src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts

```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
 		assert.equal(notebookService.getStandardKernelsForProvider('otherProvider').length, 1, 'otherProvider kernel info could not be found');
 		assert.deepEqual(notebookService.getStandardKernelsForProvider('otherProvider')[0], otherProviderRegistration.standardKernels, 'otherProviderRegistration standard kernels does not match');
 	});
+
+	test('tests that dispose() method calls dispose on underlying disposable objects exactly once', async () => {
+		await notebookService.registrationComplete;
+		notebookService.dispose();
+		assert.strictEqual(notebookService['_store']['_isDisposed'], true, `underlying disposable store object should be disposed state`);
+	});
+
+	test('Debug_verify that getProviderInstance does not throw when extensionService.whenInstalledExtensionRegistered() throws', async () => {
+		const providerId = 'providerId1';
+		createRegisteredProviderWithManager({ notebookService, providerId });
+		notebookService.registerProvider(providerId, undefined);
+		//verify method under test logs error and does not throw when extensionService.whenInstalledExtensionRegistered() throws
+		const error: Error = new Error('Extension Registration Failed');
+		extensionServiceMock.setup(x => x.whenInstalledExtensionsRegistered()).throws(error);
+
+		logServiceMock.setup(x => x.error(TypeMoq.It.isAny()))
+			.returns((_error: string | Error, ...args: any[]) => {
+				assert.ok(_error instanceof Error, `error passed into logService.error call should be an instance of ${Error}`);
+				assert.strictEqual(_error, error, `error object passed to logService.error call must be the one thrown from whenInstalledExtensionsRegistered call`);
+			})
+			.verifiable(TypeMoq.Times.once());
+		await notebookService.getOrCreateNotebookManager(providerId, URI.parse('untitled:uri1'));
+		logServiceMock.verifyAll();
+	});
+
+
+	test('verify that getProviderInstance throws when no providers are registered', async () => {
+		const methodName = 'getProviderInstance';
```

Comment:
Just get rid of these kind of vars IMO - they don't really provide any use vs just embedding in the strings directly...

Reply:
if it is used once then I agree, but in most cases, I was referring to the methodName in the verification message so wanted a const for it instead of repeating the same string.